be/src/util/internal_service_recoverable_stub.cpp

```diff
@@ -60,6 +60,67 @@ Status PInternalService_RecoverableStub::reset_channel(int64_t next_connection_g
     return Status::OK();
 }
 
+Status PInternalService_RecoverableStub::check_health() {
+#ifdef BE_TEST
+    if (_endpoint.port == 125) {
+        LOG(INFO) << "BE_TEST mode: Port 125 simulated as unhealthy for " << _endpoint;
+        return Status::InternalError("Simulated failure for port 111");
+    }
+    LOG(INFO) << "BE_TEST mode: Endpoint " << _endpoint << " simulated as healthy";
+    return Status::OK();
+#endif
+
+    // extract host
+    std::string host = butil::endpoint2str(_endpoint).c_str();
+    size_t colon_pos = host.find(':');
+    if (colon_pos != std::string::npos) {
+        host = host.substr(0, colon_pos);
+    }
+
+    // generate endpoint
+    butil::EndPoint http_endpoint;
+    if (butil::str2endpoint((host + ":" + std::to_string(config::be_http_port)).c_str(), &http_endpoint) != 0) {
```

Comment:
theoretically, it is allowed that different BE has different http port configured, this implementation will add the restriction that all the BEs/CNs have the same http port configured which should be avoided.